internal/action/process.go

```diff
@@ -0,0 +1,237 @@
+// Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+// or more contributor license agreements. Licensed under the Elastic License;
+// you may not use this file except in compliance with the Elastic License.
+
+package action
+
+import (
+	"context"
+	"fmt"
+	"strconv"
+	"strings"
+	"time"
+
+	"github.com/cenkalti/backoff/v4"
+	"github.com/elastic/e2e-testing/internal/common"
+	"github.com/elastic/e2e-testing/internal/deploy"
+	"github.com/elastic/e2e-testing/internal/utils"
+	log "github.com/sirupsen/logrus"
+)
+
+// actionWaitProcess implements operation for waiting on a process status
+type actionWaitProcess struct {
+	service deploy.ServiceRequest
+	deploy  deploy.Deployment
+	opts    map[string]string
+}
+
+// AttachActionWaitProcess action to woit for a process status on *nix like systems
+func AttachActionWaitProcess(deploy deploy.Deployment, service deploy.ServiceRequest, actionOpts map[string]string) deploy.ServiceOperatorAction {
+	return &actionWaitProcess{
+		service: service,
+		deploy:  deploy,
+		opts:    actionOpts,
+	}
+}
+
+// Run executes the command
+func (a *actionWaitProcess) Run(ctx context.Context) (string, error) {
+	timeoutFactor, _ := time.ParseDuration(a.opts["maxTimeout"])
+	exp := utils.GetExponentialBackOff(timeoutFactor)
+
+	mustBePresent := false
+	if a.opts["desiredState"] == "started" {
+		mustBePresent = true
+	}
+	retryCount := 1
+
+	processStatus := func() error {
+		occurrences, _ := strconv.Atoi(a.opts["occurrences"])
+		log.WithFields(log.Fields{
+			"desiredState": a.opts["desiredState"],
```

Comment:
I'd expose here a struct for the process options, instead of a map, so that the consumer is able to discover what to pass to the process, instead of a map of string|string.